### PR TITLE
Adds Codespell Support

### PR DESCRIPTION
**Pull request by danehans**
- Ports Contour's `codespell` spell checking support to Contour Operator.
- Fixes spelling mistakes throughout the codebase caught by `codespell`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,6 +76,13 @@ lint-golint:
 	@echo Running Go linter ...
 	@./hack/golangci-lint.sh run
 
+.PHONY: lint-codespell
+lint-codespell: CODESPELL_SKIP := $(shell cat .codespell.skip | tr \\n ',')
+lint-codespell: CODESPELL_BIN := codespell
```
Comment (stevesloka) [inline Makefile:81]: This line can be removed.

**Review by stevesloka — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (danehans) [reply Makefile]: @stevesloka commit `2985220` removes `lint-codespell: CODESPELL_BIN := codespell`. Should the same be done in projectcontour/contour?

**Comment by danehans** [comment]
Commit `95bc314` fixes merge conflicts.

File: Makefile (hunk 1 above)
Comment (stevesloka) [reply Makefile]: Yup! Looks like I missed it in the original PR but found it here. =) 

https://github.com/projectcontour/contour/pull/3141

**Review by stevesloka — APPROVED** [review]
/lgtm

**Pull request merged**